File: apps/Android/MnnLlmChat/app/src/main/java/com/alibaba/mls/api/ApplicationProvider.java
// Created by ruoyi.sjd on 2024/12/18.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.

package com.alibaba.mls.api;

import android.app.Application;

public class ApplicationProvider {
    static Application application;
    public static void set(Application application ) {
        ApplicationProvider.application = application;
    }

    public static Application get() {
        return application;
    }
}

File: apps/Android/MnnLlmChat/app/src/main/java/com/aria/mls/api/ApplicationProvider.kt
// Created by ruoyi.sjd on 2024/12/18.
// Copyright (c) 2024 Alibaba Group Holding Limited All rights reserved.
package com.alibaba.mls.api

import android.app.Application
import com.alibaba.mnnllm.android.update.UpdateChecker

object ApplicationProvider {

    var application: Application? = null
    fun set(application: Application?) {
        ApplicationProvider.application = application
        UpdateChecker.registerDownloadReceiver(application!!.applicationContext)
    }

    @JvmStatic
    fun get(): Application {
        return application!!
    }
}
